src/context/toast.test.tsx

```diff
@@ -0,0 +1,228 @@
+import { renderHook } from "@testing-library/react-hooks";
+import { render, fireEvent, waitFor, act } from "test_utils/test-utils";
+import { RequireAtMostOne } from "types/utils";
+import { RenderFakeToastContext } from "./__tests__/toast";
+import { useToastContext, ToastProvider } from "./toast";
+
+type useToastContextProps = ReturnType<typeof useToastContext>;
+
+// This type requires that the component only accepts one of the props
+type UseToastComponentProps = RequireAtMostOne<
+  {
+    [K in keyof useToastContextProps]: Parameters<useToastContextProps[K]>;
+  }
+>;
```

Comment:
This takes the return type of `useToastContext` and maps the returned `DispatchToast` object type. Into a Object with 
`Key: Function Parameter tuple` pairings.

This was necessary in order to expose these as component props below while preserving type safety.
Each callable function will be mapped to the correct function parameters as props. 

`RequireAtMostOne` also ensures that we can't pass more than one parameter into this test component. This ensures we have static type safety checks in place for this component instead of relying on implementation level validation.